packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -774,6 +789,38 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		}
 	}
 
+	/**
+	 * Delete unused storage blobs, if any. After GC runs and notifies blob manager of local blob ids that are sweep
+	 * ready, blob manager deletes them from the redirect table. This function is then called to delete the storage
+	 * ids of these blobs if they are not used anymore. A blob is unused if there is no entry from the local id to
+	 * storage id in the redirect table, i.e., all local id entries have been deleted.
+	 * @param maybeUnusedBlobIds - List of storage ids for blobs whose local id entries were just deleted and are
+	 * now candidates for deletion.
+	 */
+	private deleteUnusedStorageBlobs(maybeUnusedBlobIds: string[]) {
```

Comment:
I was confused by the code in here (esp `this.redirectTable.delete(storageId);` at the end) at first, until I realized that `maybeUnusedBlobIds` is always populated in tandem with removing a mapping from the redirect table.  Consider instead collecting the set of localIds to unmap, and having this take that in and do both parts.  Might be better encapsulation of the logic.

Reply:
Even more encapsulation would be to pull `redirectTable` into a class that hides the mapping itself and takes care of this logic.  Out of scope for this PR I think. Would have to look at how its used (and if it has other special behavior/requirements) that would further justify this encapsulation.